app/controllers/defendants_controller.rb

```diff
@@ -13,6 +13,16 @@ def show
                    defendant_path(defendant.arrest_summons_number || defendant.national_insurance_number)
   end
 
+  def remove_link
```

Comment:
i think my original hope for unlinking (it all comes back now), was to use the [laa_references_controller](https://github.com/ministryofjustice/laa-court-data-ui/blob/master/app/controllers/laa_references_controller.rb)  create and destroy actions to enable a Restful approach. @archferns would this be something you think is worth doing?

However, looking at the current state of the adaptor's API for [laa_references](https://github.com/ministryofjustice/laa-court-data-adaptor/blob/master/app/controllers/api/internal/v1/laa_references_controller.rb) and [defendants](https://github.com/ministryofjustice/laa-court-data-adaptor/blob/master/app/controllers/api/internal/v1/defendants_controller.rb#L7) it looks like this is a sensible alternative. 

Would it be clearer, however, to have the link call an update action in the controller?

very roughly and assuming routes changed
```
link_to "Unlink whatever", defendant_path(@defendant, current_user), method: :put, class: 'govuk-link'
```

Reply:
I started off with this exact approach because it was the least surprising - especially as we create references with that controller! But as you note, the adaptor API doesn't work the way I expected - and having the `laa_references_controller` try and do something with a `defendant` felt less clear than this.

It may be worth feeding back to the adaptor team that this was a surprising thing for us? I'm worried about making them make changes when there's a looming deadline, though.

I thought using a non-RESTful route might help highlight the weirdness - but I'm not sure what is best here. I'll happily change it to `#update` if you think that it doesn't hide the surprise too much!

Reply:
lets have a chat with the adaptor team, tomorrow if possible, and see if its a big ask. otherwise happy to go with this approach